traceevents/src/main/java/com/tomtom/kotlin/traceevents/Tracer.kt

```diff
@@ -674,6 +687,34 @@ public class Tracer private constructor(
             }
         }
 
+        /**
+         * Must be called after [traceEventChannel] has been updated, for example by calling
+         * [Channel.trySend] or removing elements by iterating over it.
+         *
+         * This updates whether the channel is empty. Though having to call this method to keep
+         * track of that is not ideal, there appears to be no better alternative. [Channel]s only
+         * allow receiving the elements within it (by a single collector) and do not allow observing
+         * updates to its queue.
+         */
+        @OptIn(ExperimentalCoroutinesApi::class)
+        private fun onTraceEventChannelUpdated() {
+            traceEventChannelIsEmpty.tryEmit(traceEventChannel.isEmpty)
```

Comment:
Does `tryEmit` allow it to be emitted from another context? (As this is a problem for `Flow`s)

Reply:
As far as I can tell: yes. Unlike `emit` it's not a `suspend fun` and its documentation states: "This method is **thread-safe** and can be safely invoked from concurrent coroutines without external synchronization."

If we stick with this approach, could you elaborate on what the problem would be?